examples/coappearances.cpp

```diff
@@ -0,0 +1,327 @@
+              << std::endl;
+
+    std::cout << "Characters:" << std::endl;
+    auto vertices = graph.vertices( );
+    for ( uint32_t vertex_ref = 0; vertex_ref < vertices.size( ); ++vertex_ref )
+    {
+        auto vertex = vertices[ vertex_ref ];
+        std::cout << strings + vertex.name_ref;
+        auto visitor = flatdata::make_overload(
+            [strings]( co::Nickname nickname ) {
+                std::cout << " (" << strings + nickname.ref << ")";
+            },
+            [strings]( co::Description desciption ) {
+                std::cout << ": Description: " << strings + desciption.ref;
+            },
+            [strings, vertices]( co::UnaryRelation relation ) {
+                std::cout << ": Relation(kind=" << strings + relation.kind_ref
+                          << ", to=" << strings + vertices[ relation.to_ref ].name_ref << ")";
+            },
+            [strings, vertices]( co::BinaryRelation relation ) {
+                std::cout << ": Relation(kind=" << strings + relation.kind_ref
+                          << ", to=" << strings + vertices[ relation.to_a_ref ].name_ref
+                          << ", to=" << strings + vertices[ relation.to_b_ref ].name_ref << ")";
+            } );
+        graph.vertices_data( ).for_each( vertex_ref, visitor );
+        std::cout << std::endl;
+    }
+    std::cout << std::endl;
+
+    std::cout << "Edges: " << std::endl;
+    auto edges = graph.edges( );
+    // Skip the last edge since it is a sentinel
+    for ( uint32_t edge_ref = 0; edge_ref + 1 < edges.size( ); ++edge_ref )
+    {
+        auto edge = edges[ edge_ref ];
+        std::cout << strings + vertices[ edge.a_ref ].name_ref << " meets "
+                  << strings + vertices[ edge.b_ref ].name_ref << " " << edge.count
+                  << " time(s) in chapters ";
+        // The end of the chapters assigned to this edge is the first chapter from the next edge.
+        // This is a typical trick when storing ranges. That's why a sentinel was added to edges.
+        uint32_t next_edge_ref = edge_ref + 1;
+        uint32_t chapters_begin = edge.first_chapter_ref;
+        uint32_t chapters_size = edges[ next_edge_ref ].first_chapter_ref - chapters_begin;
+        auto chapters = graph.chapters( ).slice( chapters_begin, chapters_size );
+        for ( size_t chapter_ref = 0; chapter_ref < chapters.size( ); ++chapter_ref )
+        {
+            auto chapter = graph.chapters( )[ chapter_ref ];
+            std::cout << chapter.major.as< uint32_t >( ) << "." << chapter.minor.as< uint32_t >( )
+                      << ( chapter_ref + 1 < chapters.size( ) ? ", " : "." );
+        }
+        std::cout << std::endl;
+    }
+}
+
+static const char* USAGE = 1 + R"_(
+Usage:
+  coappearances convert <input.json> <output_folder>
+  coappearances read <output_folder>)_";
+
+int
+main( int argc, char const* argv[] )
+{
+    if ( argc < 3 )
```

Comment:
Might make sense to use `gflags` or similar library to make this leaner. As an example, this can afford extra dependency IMO.

Reply:
Used docopt. Seems both to be an overkill. I think I will leave it as is since other examples are also made in the same way. If we want to change it we should make a separate review changing all examples.